docs/learn/dapp-developer/read-asset-data.md

```diff
@@ -266,6 +266,286 @@ console.log(assetMetadata);
+  INTERFACE_IDS.LSP8IdentifiableDigitalAsset,
+);
+
+async function fetchTokenIdMetadata() {
+  if (!isLSP8) {
+    console.log('Asset is not an LSP8.');
+    return;
+  }
+
+  // Token ID as Bytes32 value (1)
+  const tokenID =
+    '0x0000000000000000000000000000000000000000000000000000000000000001';
+
+  // Get the encoded asset metadata
+  const tokenIdMetadata = await myAssetContract.methods.getDataForTokenId(
+    tokenID,
+    ERC725YDataKeys.LSP4['LSP4Metadata'],
+  );
+
+  const erc725js = new ERC725(lsp4Schema);
+
+  // Decode the metadata
+  const decodedMetadata = erc725js.decodeData([
+    {
+      keyName: 'LSP4Metadata',
+      value: tokenIdMetadata,
+    },
+  ]);
+
+  console.log(decodedMetadata);
+}
+
+fetchTokenIdMetadata();
+```
+
+  </TabItem>
+
+</Tabs>
+
+<Details>
+<summary>Show result</summary>
+
+```js
+{
+    "key": "0x9afb95cacc9f95858ec44aa8c3b685511002e30ae54415823f406128b85b238e",
+    "name": "LSP4Metadata",
+    "value": {
+      "verification": {
+        "method": "keccak256(utf8)",
+        "data": "0x9eefcdac3d60500619b075273c0371a6633d8f531179c882facd4f991281c658"
+      },
+      "url": "ipfs://QmeKNiTr4xfdHDUinGmYC4osu1ZHoFHDDj87WBSX5z4k7x"
+    }
+  }
+```
+
+</Details>
+
+### Get JSON Content
+
+Based on the decoded metadata from the contract, you can continue retrieving the actual contents of the [`Verifiable URI`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-2-ERC725YJSONSchema.md#verifiableuri). Usually, you can retrieve the [`LSP8TokenMetadataBaseURI`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#lsp8tokenmetadatabaseuri) from the asset, indicating which file storage is used. This address can then be placed in front of the content ID to get the full link to the file:
+
+```js
+async function fetchTokenIdMetadata() {
+  // ...
+
+  // Get the BaseURI of the storage provider
+  let tokenBaseURI = await myAsset.getData(
+    ERC725YDataKeys.LSP8['LSP8TokenMetadataBaseURI'],
+  );
+  console.log('BaseURI: ', tokenBaseURI);
+
+  // Get the contentID or file link
+  let contentID = decodedMetadata[0].value.url;
+  console.log('ContentID: ', contentID);
+}
+
+fetchTokenIdMetadata();
+```
+
+:::info
+
+The file address within the fetched and decoded `LSP4 Metadata` of the token ID might already contain the full link of the file, without using the base provider. If you are checking assets in production, please check if the content is a valid URL beforehand.
+
+:::
+
+:::caution
+
+In case the [`LSP8TokenMetadataBaseURI`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#lsp8tokenmetadatabaseuri) was not set, you will have to fall back to the gateway indicated within the `contentID`. In this example, we will use the IPFS gateway of LUKSO, as the `contentID` starts with `ipfs://`, meaning that it was uploaded using the [Interplanetary File System](https://ipfs.tech/).
```

Comment:
Sorry, i still think we are missing something. We dont have a section showing how to get the metadata of a specific tokenId if the asset uses `LSP8TokenMetadataBaseURI`.

Instead of fetching LSP4Metadata per tokenId as in the `fetchTokenIdMetadata` function, you would need to fetch the base URI retrieved from `LSP8TokenMetadataBaseURI` to construct the metadata url as `https://mybase.uri/TOKEN_ID` which can be used for all assets

Reply:
Thanks for bearing with me.  🙏🏻 I just talked back to Daniel, and we should indeed split Metadata and BaseURI up into separate sections instead of combining them. 

I thought they were used for the same purpose and could be combined somehow. I will get back tomorrow and check for a good example asset. The current one does not have a BaseURI set. 🙂

Reply:
perfect thanks!